pkg/agent/proxy/proxier_test.go

```diff
@@ -1962,37 +1962,43 @@ func TestLoadBalancerRemove(t *testing.T) {
 	})
 }
 
-func testClusterIPNoEndpoint(t *testing.T, svcIP net.IP, isIPv6 bool) {
+func testClusterIPNoEndpoint(t *testing.T, bindingProtocol binding.Protocol, isIPv6 bool) {
 	ctrl := gomock.NewController(t)
 	mockOFClient, mockRouteClient := getMockClients(ctrl)
 	groupAllocator := openflow.NewGroupAllocator()
-	fp := newFakeProxier(mockRouteClient, mockOFClient, nil, groupAllocator, isIPv6)
+	apiProtocol := getAPIProtocol(bindingProtocol)
+	// Create a ServicePort with a specific protocol, avoiding using the global variable 'svcPortName' which is set to TCP protocol.
+	svcPortName := makeSvcPortName("ns", "svc", strconv.Itoa(svcPort), apiProtocol)
+	svcIP := svc1IPv4
+	if isIPv6 {
+		svcIP = svc1IPv6
+	}
+	fp := newFakeProxier(mockRouteClient, mockOFClient, nil, groupAllocator, isIPv6, withCleanupStaleUDPSvcConntrack)
 
-	svc := makeTestClusterIPService(&svcPortName, svcIP, nil, int32(svcPort), corev1.ProtocolTCP, nil, nil, false, nil)
-	updatedSvc := makeTestClusterIPService(&svcPortName, svcIP, nil, int32(svcPort+1), corev1.ProtocolTCP, nil, nil, false, nil)
+	svc := makeTestClusterIPService(&svcPortName, svcIP, nil, int32(svcPort), apiProtocol, nil, nil, false, nil)
+	updatedSvc := makeTestClusterIPService(&svcPortName, svcIP, nil, int32(svcPort+1), apiProtocol, nil, nil, false, nil)
 	makeServiceMap(fp, svc)
 	makeEndpointSliceMap(fp)
 
-	protocol := binding.ProtocolTCP
-	if isIPv6 {
-		protocol = binding.ProtocolTCPv6
-	}
 	mockOFClient.EXPECT().InstallServiceGroup(binding.GroupIDType(1), false, []k8sproxy.Endpoint{}).Times(1)
 	mockOFClient.EXPECT().InstallServiceFlows(&antreatypes.ServiceConfig{
 		ServiceIP:          svcIP,
 		ServicePort:        uint16(svcPort),
-		Protocol:           protocol,
+		Protocol:           bindingProtocol,
 		TrafficPolicyLocal: false,
 		ClusterGroupID:     1,
 	}).Times(1)
 	fp.syncProxyRules()
 	assert.Contains(t, fp.serviceInstalledMap, svcPortName)
 
 	mockOFClient.EXPECT().UninstallServiceFlows(svcIP, uint16(svcPort), gomock.Any()).Times(1)
+	if needClearConntrackEntries(bindingProtocol) {
+		mockRouteClient.EXPECT().ClearConntrackEntryForService(svcIP, uint16(svcPort), nil, bindingProtocol).Times(1)
```

Comment:
I will double-check this.

Reply:
I have created a UDP Service with no Endpoints, when I run the following command ` nc -u 10.100.159.34 80`,

I got the result of tcpdump on `antrea-gw0`:

```
tcpdump -i antrea-gw0 host 10.100.159.34 -nve
tcpdump: listening on antrea-gw0, link-type EN10MB (Ethernet), capture size 262144 bytes
16:15:29.966821 62:bd:93:1a:81:21 > aa:bb:cc:dd:ee:ff, ethertype IPv4 (0x0800), length 43: (tos 0x0, ttl 64, id 49870, offset 0, flags [DF], proto UDP (17), length 29)
    10.10.0.1.36158 > 10.100.159.34.80: UDP, length 1
16:15:29.967399 aa:bb:cc:dd:ee:ff > 62:bd:93:1a:81:21, ethertype IPv4 (0x0800), length 70: (tos 0x0, ttl 128, id 10040, offset 0, flags [none], proto ICMP (1), length 56)
    10.100.159.34 > 10.10.0.1: ICMP host 10.100.159.34 unreachable - admin prohibited, length 36
        (tos 0x0, ttl 64, id 49870, offset 0, flags [DF], proto UDP (17), length 29)
    10.10.0.1.36158 > 10.100.159.34.80: UDP, length 1
```

I also got a conntrack entry:

```
udp      17 24 src=10.10.0.1 dst=10.100.159.34 sport=58044 dport=80 [UNREPLIED] src=10.100.159.34 dst=10.10.0.1 sport=80 dport=58044 mark=0 use=1
```

For TCP, I didn't find a conntrack entry for a Service without Endpoints.

Reply:
Thanks for checking @hongliangl. But my question was specifically about the Antrea ct zone (65520). I assume that we don't commit any ct entry there in that case. In your example, the UDP ct entry seems to be in the default zone, and I assume it was committed by the host.

Reply:
@antoninbas You just reminded me. I got the conntrack results by filtering with Service IP 10.100.159.34, and there is no corresponding contrack entries in CT zone 65520.  Though it is harmless to delete the non-existent entries, I think we can fix this in another PR.

Reply:
It is harmless. I think from a code perspective, it is better to avoid handling that special case differently. It just makes the code more complex. I guess a comment (at the place where the function is called though) is always useful though. But it's unrelated to this specific PR.